src/lib/configs/project/dockerCompose.ts

```diff
@@ -0,0 +1,315 @@
+        FLUENCE_ROOT_KEY_PAIR__PATH: `/run/secrets/${name}`,
+      },
+      command: [
+        "--aqua-pool-size=5",
+        `-t=${tcpPort}`,
+        `-w=${webSocketPort}`,
+        "--external-maddrs",
+        `/dns4/${name}/tcp/${tcpPort}`,
+        `/dns4/${name}/tcp/${webSocketPort}/ws`,
+        "--allow-private-ips",
+        tcpPort === TCP_PORT_START
+          ? "--local"
+          : `--bootstraps=/dns/${bootStrapNoxName}/tcp/${TCP_PORT_START}`,
+      ],
+      depends_on: [IPFS_CONTAINER_NAME],
+      secrets: [name],
+    },
+  ];
+}
+
+async function genDockerCompose(
+  providerConfig: ProviderConfigReadonly,
+): Promise<LatestConfig> {
+  const secretKeys = await genSecretKeysOrReturnExisting(
+    Object.keys(providerConfig.computePeers),
+  );
+
+  const [bootStrapNoxNameAndKey, ...restSecretKeys] = secretKeys;
+
+  assert(
+    bootStrapNoxNameAndKey !== undefined,
+    "At least one nox is required to generate a docker-compose.yml",
+  );
+
+  const { name: bootStrapNoxName } = bootStrapNoxNameAndKey;
+
+  return {
+    version: "3",
+    services: {
+      [CHAIN_CONTAINER_NAME]: {
+        image: versions.chain,
+        pull_policy: "always",
+        ports: [`${CHAIN_PORT}:${CHAIN_PORT}`],
+      },
+      [IPFS_CONTAINER_NAME]: {
+        image: "ipfs/go-ipfs",
+        pull_policy: "always",
+        ports: [`${IPFS_PORT}:${IPFS_PORT}`, "4001:4001"],
+        environment: {
+          IPFS_PROFILE: "server",
+        },
+        volumes: [`./${IPFS_CONTAINER_NAME}/:/container-init.d/`],
+      },
+      ...Object.fromEntries([
+        genNox({
+          name: bootStrapNoxName,
+          tcpPort: TCP_PORT_START,
+          webSocketPort: WEB_SOCKET_PORT_START,
+          bootStrapNoxName,
+        }),
+      ]),
+      ...Object.fromEntries(
+        restSecretKeys.map(({ name }, index) => {
+          return genNox({
+            name,
+            tcpPort: TCP_PORT_START + 1 + index,
+            webSocketPort: WEB_SOCKET_PORT_START + 1 + index,
+            bootStrapNoxName,
+          });
+        }),
+      ),
+    },
+    secrets: Object.fromEntries(
+      secretKeys.map(({ name, path: file }) => {
+        return [name, { file }];
+      }),
+    ),
+  };
+}
+
+async function genDefaultDockerCompose(
+  providerConfig: ProviderConfigReadonly,
+): Promise<GetDefaultConfig> {
+  const def = await genDockerCompose(providerConfig);
+
+  return () => {
+    return yamlDiffPatch("", {}, def);
+  };
+}
+
+const migrations: Migrations<Config> = [];
+
+type Config = ConfigV0;
+type LatestConfig = ConfigV0;
+export type DockerComposeConfig = InitializedConfig<LatestConfig>;
+export type DockerComposeConfigReadonly =
+  InitializedReadonlyConfig<LatestConfig>;
+
+const initConfigOptions = {
+  allSchemas: [configSchemaV0],
+  latestSchema: configSchemaV0,
+  migrations,
+  name: DOCKER_COMPOSE_FILE_NAME,
+  getConfigOrConfigDirPath: getFluenceDir,
+};
+
+export async function initNewDockerComposeConfig(
```

Comment:
let's add some docs please, I don't understand the difference between `initNewDockerComposeConfig` and `initNewReadonlyDockerComposeConfig`

Reply:
let's document that in code plz

Reply:
I will. It's natural to code until it's clear that what you are doing is pretty hard and involved and then to refactor and document. It would be a waste if I did it from the begining  https://linear.app/fluence/issue/DXJ-509/refactor-configs-in-cli